### DISCUSSION TITLE
Signature of `TARGET_LINK_LIBRARIES` in CMake macros

### DISCUSSION THREAD

**Discussion by davidscn** [pybind/pybind11#3340] [Q&A]
Hello everyone,

we recently revived an older project using still pybin11 v1.8.1 and updated the pybind11 version in a first step. Doing so, we ran into the following issue

```bash
  The keyword signature for target_link_libraries has already been used with
  the target "...".  All uses of target_link_libraries with
  a target must be either all-keyword or all-plain.

   * pybind11/tools/pybind11Tools.cmake:162 (target_link_libraries)
   * pybind11/tools/pybind11Tools.cmake:194 (target_link_libraries)
```
which results from the fact that the signature of `TARGET_LINK_LIBRARIES` was changed. I can easily change the signature in the respective CMake file, but there are other dependencies using macros with a different signature so that I cannot easily change the signature in all relevant places. Is it possible to use pybind11 without the keyword signature or are you aware of any other way around this issue?

_FYI_ @renefritze

**Comment by chunlai67** [comment]
put keyword `PRIVATE` in your TARGET_LINK_LIBRARIES  would do the trick
``` cake
TARGET_LINK_LIBRARIES(<target> PRIVATE <lib2linkagainst>)
```

**Comment by davidscn** [comment]
Thanks for the suggestion, I'm aware that the same signatures work. However, I have two distinct third party projects using conflicting signatures. Hence, it's not my code and I can't easily modify it, which lead to the discussion here.
